sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -0,0 +1,118 @@
+# coding=utf-8
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import datetime as dt
+import uuid
+import base64
+
+try:
+    from typing import TYPE_CHECKING
+except ImportError:
+    TYPE_CHECKING = False
+
+if TYPE_CHECKING:
+    from typing import Any, Dict
+
+__all__ = ["CloudEvent"]
+
+def _decode(content):
+    try:
+        return base64.b64decode(content.encode('utf-8'))
+    except (ValueError, TypeError) as error:
+        # ValueError for Python 3, TypeError for Python 2
+        raise ValueError(
+            message="Data is not valid base 64.",
+            error=error)
+
+class CloudEvent(object):   #pylint:disable=too-many-instance-attributes
+    """Properties of the CloudEvent 1.0 Schema.
+
+    All required parameters must be populated in order to send to Azure.
+    If data is of binary type, data_base64 can be used alternatively. Note that data and data_base64
+    cannot be present at the same time.
+
+    :param source: Required. Identifies the context in which an event happened. The combination of id and source must
+     be unique for each distinct event. If publishing to a domain topic, source must be the domain name.
+    :type source: str
+    :param type: Required. Type of event related to the originating occurrence.
+    :type type: str
+    :keyword data: Optional. Event data specific to the event type. If data is of bytes type, it will be sent
+     as data_base64 in the outgoing request.
+    :type data: object
+    :keyword time: Optional. The time (in UTC) the event was generated, in RFC3339 format.
+    :type time: ~datetime.datetime
+    :keyword dataschema: Optional. Identifies the schema that data adheres to.
+    :type dataschema: str
+    :keyword datacontenttype: Optional. Content type of data value.
+    :type datacontenttype: str
+    :keyword subject: Optional. This describes the subject of the event in the context of the event producer
+     (identified by source).
+    :type subject: str
+    :keyword specversion: Optional. The version of the CloudEvent spec. Defaults to "1.0"
+    :type specversion: str
+    :keyword id: Optional. An identifier for the event. The combination of id and source must be
```

Comment:
We don't need explicit "Optional." for keyword arguments. Right? :)

Reply:
hmm, not sure if it's a good practice - but there can certainly be "required" keyword only params (hopefully not in our libraries,  but still possible) - thought no harm in mentioning it's optional :)